### PR TITLE
Place `ttrpc` files in `OUT_DIR`

### PR DESCRIPTION
**Pull request by jprendes**
This PR places the generated `ttrpc` files in `OUT_DIR`, and makes the files generation unconditional.
This avoid commiting the generated files to the repo.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -121,14 +121,6 @@ You can auto-fix most styles using
 make fix
 ```
 
-## Updating protobuf files
-
-Ensure protoc and dev tools have been installed.  Run `make build` or to just generate the protos:
-
-```
-cargo build -p containerd-shim-wasm --no-default-features --features generate_bindings
```
[the remark this answers is not in the document]
Comment (jprendes) [reply CONTRIBUTING.md:129]: The actual location of `OUT_DIR` is an implementation detail of `cargo`.

Do they need to be discoverable?
The `.proto` file is discoverable and what we should care about.

As I see it, the generated files are just an artifact.
They are not really intended for human consumption, we don't even lint them.

[the remark this answers is not in the document]
Comment (jsturtevant) [reply CONTRIBUTING.md:129]: This is similiar to how rust-extension project does it as well https://github.com/containerd/rust-extensions/blob/2f0ffb98b5dcf5e5473775bcf9e49abf0ccbc9f0/crates/shim-protos/src/shim.rs#L17-L19

**Comment by jsturtevant** [comment]
LGTM

**Review by Mossaka — DISMISSED** [review]
lgtm!

**Comment by Mossaka** [comment]
It seems like this needs a rebase.

**Comment by cpuguy83** [comment]
Why would we not commit the files to the repo?

**Comment by jprendes** [comment]
> Why would we not commit the files to the repo?

It's generally good practice not including generated files.

They are a distraction and can be confusing.
It adds noise to the makefile, which is currently in charge of generating these files before building.
It adds a seemingly useless (for downstream consumers) feature to the crate.
Also contributors might think it's ok to modify them manually, just for their changes to be lost the next time they are generated.
Their content is an implementation detail of `ttrpc-rust`, not particularly human friendly code.
Also, if we move to shims v3, are we going to have the `ttrcp-rust` generated files for `ttrcp` and the `tonic` generated files for the `grpc` implementation?

There's also no benefit to having them in the repo.
Build time impact is negligible (they are generated only once, cargo keeps track of this).
It does need protoc installed, but we already need it for `rust-extensions`.

**Review by Mossaka — APPROVED** [review]
LGTM!

**Comment by cpuguy83** [comment]
This is fine I guess.
I expect it to break code lookups since those types are no longer there, but I guess the files are being generated when it compiles them for code analysis.

**Comment by jprendes** [comment]
Lookup works fine with `rust-analyzer` on `vscode`.
As @jsturtevant mentioned, it's the same approach as in [containerd/rust-extensions](https://github.com/containerd/rust-extensions/)

**Pull request merged**